lib/utils/unstringifyBools.js

```diff
@@ -0,0 +1,19 @@
+//
+// unstringifyBools
+// changes "true" and "false" strings into actual boolean values throughout an array
+//
+
+/**
+ * unstringifyBools
+ * convert boolean inputs from strings into actual bools within a flat array
+ * @param {array} inputArray  the array to process
```

Comment:
While it is true this code should work on both arrays and objects, we are expecting objects to be passed in here, right?

Reply:
Haha, yes you're right.  I think the syntax on between using the two is a bit fuzzier in Javascript compared to other languages.  I'll update the comment and input names.